src/ToricVarieties/AlgebraicCycles/attributes.jl

```diff
@@ -0,0 +1,218 @@
+########################
+# Attributes of algebraic cycles
+########################
+
+@doc Markdown.doc"""
+    toric_variety(ac::AlgebraicCycle)
+
+Return the normal toric variety of the algebraic cycle `ac`.
+
+# Examples
+```jldoctest
+julia> dP2 = del_pezzo(2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> d = ToricDivisor(dP2, [1,2,3,4,5])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> ac = AlgebraicCycle(d)
+An algebraic cycle on a normal toric variety given by 6*V(x3)+1*V(e1)+7*V(e2)
+
+julia> toric_variety(ac)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+```
+"""
+toric_variety(ac::AlgebraicCycle) = ac.v
+export toric_variety
+
+
+@doc Markdown.doc"""
+    coefficients(ac::AlgebraicCycle)
+
+Return the coefficients of the algebraic cycle `c`.
+
+# Examples
+```jldoctest
+julia> dP2 = del_pezzo(2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> d = ToricDivisor(dP2, [1,2,3,4,5])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> cc = AlgebraicCycle(d)
+An algebraic cycle on a normal toric variety given by 6*V(x3)+1*V(e1)+7*V(e2)
+
+julia> coefficients(cc)
+3-element Vector{fmpq}:
+ 6
+ 1
+ 7
+```
+"""
+coefficients(ac::AlgebraicCycle) = [coefficient_ring(toric_variety(ac))(k) for k in coefficients(polynomial(ac).f)]
```

Comment:
So... "the" coefficients of an AlgebraicCycle are not actually well-defined, but depend on some chosen representative `p.f` ?

I think this is not very helpful unless the documentation makes clearer what this is all about? Or is this really the way these objects are treated in the literature?

Reply:
I can see how this could be confusing. I have just added more details.